pkg/interactor/data.go

```
@@ -0,0 +1,59 @@
+package interactor
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/CESARBR/knot-cloud-storage/pkg/data"
+	. "github.com/CESARBR/knot-cloud-storage/pkg/entities"
+)
+
+var dataStore = data.NewDataStore()
+
+type DataInteractor struct{}
+
+func NewDataInteractor() *DataInteractor {
+	return &DataInteractor{}
+}
+
+func (d *DataInteractor) GetAll(order int, skip int, take int, startDate time.Time, finishDate time.Time) ([]Data, error) {
+
+	selectOrder := "timestamp"
+	if order == -1 {
+		selectOrder = "-timestamp"
+	}
+
+	data, err := dataStore.Get(selectOrder, skip, take, startDate, finishDate)
+
+	return data, err
+}
+
+func (d *DataInteractor) GetByID(id string, order int, skip int, take int, startDate time.Time, finishDate time.Time) ([]Data, error) {
+
+	selectOrder := "timestamp"
+	if order == -1 {
+		selectOrder = "-timestamp"
+	}
+
+	s, _ := strconv.ParseInt(id, 10, 64)
```

I think you should handle the returned error.

---

Never discard errors :stuck_out_tongue: 
>_"Do not discard errors using `_` variables. If a function returns an error, check it to make sure the function succeeded. Handle the error, return it, or, in truly exceptional situations, panic."_

[by CodeReviewComments](https://github.com/golang/go/wiki/CodeReviewComments#handle-errors)

---

@lcbm I can't find the "chicote" emoji here, uasuhuahuas.

---

Done!